.github/workflows/complex.yaml

```diff
@@ -0,0 +1,120 @@
+name: Complex test environment
+# complex test environment with multiple k8s nodes, disks, and network options
+on:
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+defaults:
+  run:
+    # reference: https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#using-a-specific-shell
+    shell: bash --noprofile --norc -eo pipefail -x {0}
+
+jobs:
+  complex:
+    if: ${{ github.event_name == 'pull_request' && github.ref != 'refs/heads/master' && !contains(github.event.pull_request.labels.*.name, 'skip-ci') }}
+    runs-on: macos-10.15
+    steps:
+    - name: checkout
+      uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: setup golang
+      uses: actions/setup-go@v2
+      with:
+        go-version: 1.16
+
+    - name: install prerequisites
+      run: |
+        brew install coreutils docker minikube tree virtualbox
+
+    # manusa/actions-setup-minikube only works in ubuntu, but macos is the only github runner that
+    # supports nested virtualization
+    # macos VM specs: 3 cores of CPU, 14G RAM, 14G SSD space
+    - name: setup minikube cluster
+      run: |
+        NUM_DISKS_PER_VM=3
+        EXTRA_DISK_SIZE_MB=10000
+
+        time minikube start --driver=virtualbox --memory=6g --nodes=3
```

Comment:
See:

```
Hardware specification for macOS virtual machines:

3-core CPU
14 GB of RAM memory
14 GB of SSD disk space
```

So we can bump the memory a bit I think.

Reply:
6G from 3 nodes is 18G. I don't think we want to over-provision the host by too much, do we?

Reply:
Oh right, 3 nodes. Sounds good to keep it like this.